crates/environ/src/module.rs

```diff
@@ -55,18 +55,19 @@ pub enum MemoryStyle {
 impl MemoryStyle {
     /// Decide on an implementation style for the given `Memory`.
     pub fn for_memory(memory: Memory, tunables: &Tunables) -> (Self, u64) {
-        if let Some(maximum) = memory.maximum {
-            if maximum <= tunables.static_memory_bound {
-                // A heap with a declared maximum can be immovable, so make
-                // it static.
-                assert_ge!(tunables.static_memory_bound, memory.minimum);
-                return (
-                    Self::Static {
-                        bound: tunables.static_memory_bound,
-                    },
-                    tunables.static_memory_offset_guard_size,
-                );
-            }
+        // A heap with a maximum that doesn't exceed the static memory bound specified by the
+        // tunables make it static.
+        //
+        // If the module doesn't declare explicit maximum treat it as 4GiB.
+        let maximum = memory.maximum.unwrap_or(0x1_0000);
```

Comment:
This can use `WASM_MAX_PAGES` instead of hard-coding the constant value.

Reply:
Ah, great!